### PR TITLE
fix(gatsby-plugin-mdx): pass the correct options to setParserPlugins

### PR DESCRIPTION
**Pull request by d4rekanguok**
## Description

Hey folks!

In `gatsby-plugin-mdx`, when `setParserPlugins` is called, it was passed in the wrong options<sup> [source](https://github.com/gatsbyjs/gatsby/blob/561d33e2e491d3971cb2a404eec9705a5a493602/packages/gatsby-plugin-mdx/gatsby/source-nodes.js#L93)</sup>:

```
for (let plugin of options.gatsbyRemarkPlugins) {
  ...
    for (let parserPlugin of requiredPlugin.setParserPlugins(
        plugin.pluginOptions   // <--- undefined
      )) { ... }
```

Instead of `plugin.pluginOptions`, it should have been `plugin.options`.

## Reproduction of bug

https://github.com/d4rekanguok/gatsby-bug-repro-18243

### CODE REVIEW

**Comment by pieh** [comment]
Seems like this is pretty much copy & paste from remark plugin - https://github.com/gatsbyjs/gatsby/blob/68dbb6d8eaeeb07613a810d3f96cd7ec3636e2cb/packages/gatsby-transformer-remark/src/extend-node-type.js#L128-L142

Which means if that's a bug we have it in both mdx and remark.

Do you have any reproduction so we could verify this change quickly?

**Comment by d4rekanguok** [comment]
Hi @pieh! I actually discovered this while debugging a remark plugin that worked with gatsby-transformer-remark but not gatsby-plugin-mdx, so I think the issue only exists with mdx.

I can quickly make & share a reproduction repo in an hour or so

**Comment by d4rekanguok** [comment]
@pieh I've just put this repro together: 

https://github.com/d4rekanguok/gatsby-bug-repro-18243

The step to reproduce is detailed in the readme.

I think it might have something to do with the fact that remark plugins for mdx are passed in via options as `gatsbyRemarkPlugins`, while with remark it's `plugins`. I'm guessing Gatsby might have a recursive system that store plugin options as `pluginOptions`, but only recognize plugins set via the `plugins` attribute?

**Comment by pieh** [comment]
> think it might have something to do with the fact that remark plugins for mdx are passed in via options as `gatsbyRemarkPlugins`, while with remark it's `plugins`. I'm guessing Gatsby might have a recursive system that store plugin options as `pluginOptions`, but only recognize plugins set via the `plugins` attribute?

Oh yeah, that would explain the difference.

You are correct - `plugins` field in `options` does have special handling, and we do some transformation, so if `gatsbyRemarkPlugins` is used, this transformation won't be done.

**Comment by d4rekanguok** [comment]
Oh in that case, instead of fixing this like I did in this PR, maybe it's better to also handle `gatsbyRemarkPlugins` specially to keep the plugin chain going?

This might be a really edge case however, since gatsby-plugin-mdx has been released for a long time & no one has yet filed a bug on this

**Comment by pieh** [comment]
I think this is fine. Other subplugin API use `options` instead of `pluginOptions` in mdx, so I think this is fine for now.

mdx (using `plugin.options`):
https://github.com/gatsbyjs/gatsby/blob/b4e3c5eb0192832403beebad5a44db08efbfa0ca/packages/gatsby-plugin-mdx/utils/get-source-plugins-as-remark-plugins.js#L53-L73


remark (using `plugin.pluginOptions`):

https://github.com/gatsbyjs/gatsby/blob/b4e3c5eb0192832403beebad5a44db08efbfa0ca/packages/gatsby-transformer-remark/src/extend-node-type.js#L216-L235

File: packages/gatsby-plugin-mdx/gatsby/source-nodes.js [hunk 1]
```diff
@@ -90,7 +90,7 @@ module.exports = (
     debug(`required`, plugin)
     if (_.isFunction(requiredPlugin.setParserPlugins)) {
       for (let parserPlugin of requiredPlugin.setParserPlugins(
-        plugin.pluginOptions
+        plugin.options
```
Comment (pieh) [inline packages/gatsby-plugin-mdx/gatsby/source-nodes.js]: Let's ensure it's not undefined so plugins don't have to check that
```suggestion
        plugin.options || {} 
```
( Same as in https://github.com/gatsbyjs/gatsby/blob/b4e3c5eb0192832403beebad5a44db08efbfa0ca/packages/gatsby-plugin-mdx/utils/get-source-plugins-as-remark-plugins.js#L67 )

Comment (d4rekanguok) [reply packages/gatsby-plugin-mdx/gatsby/source-nodes.js]: Good call, I've actually just run into this issue a few days ago https://github.com/gatsbyjs/gatsby/issues/18135#event-2688457725

**Comment by d4rekanguok** [comment]
A few of my PRs is failing this unit_tests_node12:

> FAIL  peril/tests/validate-yaml/validate-yaml-starters.test.ts
>  ● a new PR › Doesn't allow non GitHub repos

I'm not submitting new starters though, so I'm not sure why it's failing... any ideas? @pieh

**Comment by pieh** [comment]
> I'm not submitting new starters though, so I'm not sure why it's failing... any ideas? @pieh

Oh yeah, this has something with recent Node 12 release (I think) - we "fixed" this (or rather workarounded it) in https://github.com/gatsbyjs/gatsby/pull/18247, so I'll just merge master in this PR and that should fix that failing test

**Comment by pieh** [comment]
Published `gatsby-plugin-mdx@1.0.49`

**Pull request merged**